ldes-fragmentisers/ldes-fragmentisers-substring/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/fragmentisers/substring/processor/SubstringPreProcessor.java

```diff
@@ -0,0 +1,67 @@
+package be.vlaanderen.informatievlaanderen.ldes.server.fragmentisers.substring.processor;
+
+import be.vlaanderen.informatievlaanderen.ldes.server.domain.tree.member.entities.Member;
+import be.vlaanderen.informatievlaanderen.ldes.server.fragmentisers.substring.config.SubstringConfig;
+
+import java.text.Normalizer;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.IntStream;
+
+import com.apicatalog.jsonld.StringUtils;
+
+import static be.vlaanderen.informatievlaanderen.ldes.server.fragmentisers.substring.constants.SubstringConstants.ROOT_SUBSTRING;
+
+public class SubstringPreProcessor {
```

Comment:
Personally, I'm not in favor of adding behavior together under a general name such as 'PreProcessor'. I'd prefer to keep these split and have two separate unit tests for the functionality.

Note also that substringconfig is an attribute of this class, but not used for bucketizing or tokenizen.

Reply:
My reluctance is increased by seeing the @Nested annotations in the test